### PR TITLE
[CK_TILE] Move hipmalloc/memcpy calls out of gpu reference gemm

### PR DESCRIPTION
**Pull request by aledudek**

### CODE REVIEW

File: example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc [hunk 1]
```diff
@@ -208,6 +255,35 @@ int run_batched_gemm_example_with_layouts(int argc,
...
+            std::cerr << "Error free the B memory: " << hipGetErrorString(errB) << std::endl;
+        }
+
+        errC = hipFree(d_C);
+        if(errC != hipSuccess)
+        {
+            std::cerr << "Error free the C memory: " << hipGetErrorString(errC) << std::endl;
+        }
+
+        //// TEST
```
Comment (bartekxk) [inline example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc]:
```suggestion
```

File: example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc [hunk 2]
```diff
@@ -208,6 +255,35 @@ int run_batched_gemm_example_with_layouts(int argc,
...
+                         d_C,
+                         batch_count * M * N * sizeof(CDataType),
+                         hipMemcpyDeviceToHost);
+        if(errC != hipSuccess)
+        {
+            std::cerr << "Error copying C to device: " << hipGetErrorString(errC) << std::endl;
+        }
+
+        errA = hipFree(d_A);
+        if(errA != hipSuccess)
```
Comment (bartekxk) [inline example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc]: Can we add some macro for these checks? (I suppose that it exist somewhere)

File: example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc [hunk 3]
```diff
@@ -188,15 +188,62 @@ int run_batched_gemm_example_with_layouts(int argc,
         c_m_n_gpu_ref.SetZero();
         c_m_n_gpu_buf_ref.SetZero();
 
+        //// TEST
```
Comment (bartekxk) [inline example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc]:
```suggestion
```

File: example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc [hunk 4]
```diff
@@ -188,15 +188,62 @@ int run_batched_gemm_example_with_layouts(int argc,
...
+        ADataType* d_A;
+        BDataType* d_B;
+        CDataType* d_C;
+
+        hipError_t errA = hipMalloc(&d_A, batch_count * M * K * sizeof(ADataType));
+        hipError_t errB = hipMalloc(&d_B, batch_count * N * K * sizeof(BDataType));
+        hipError_t errC = hipMalloc(&d_C, batch_count * M * N * sizeof(CDataType));
+        if(errA != hipSuccess)
```
Comment (bartekxk) [inline example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc]: Can we add some macro for these checks? (I suppose that it exist somewhere)

File: example/ck_tile/03_gemm/run_gemm_example.inc [hunk 5]
```diff
@@ -161,14 +161,82 @@ int run_gemm_example_with_layouts(int argc,
...
         c_m_n_gpu_buf_ref.SetZero();
 
+        ADataType* d_A;
+        BDataType* d_B;
+        CDataType* d_C;
+
+        hipError_t errA = hipMalloc(&d_A, M * K * sizeof(ADataType));
+        hipError_t errB = hipMalloc(&d_B, N * K * sizeof(BDataType));
+        hipError_t errC = hipMalloc(&d_C, M * N * sizeof(CDataType));
+        if(errA != hipSuccess)
```
Comment (bartekxk) [inline example/ck_tile/03_gemm/run_gemm_example.inc]: Can we create similar macro to include/ck/host_utility/hip_check_error.hpp:10 ?

**Review by bartekxk — COMMENTED** [review]
LGTM just minor comments

File: example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc (hunk 4 above)
Comment (aledudek) [reply example/ck_tile/16_batched_gemm/run_batched_gemm_example.inc]: You're right, we have a function for that, fixed in latest change

File: example/ck_tile/03_gemm/run_gemm_example.inc (hunk 5 above)
Comment (aledudek) [reply example/ck_tile/03_gemm/run_gemm_example.inc]: You're right, we have a function for that, fixed in latest change

Comment (bartekxk) [reply example/ck_tile/03_gemm/run_gemm_example.inc]: You forgot to add these change also in this file

Comment (aledudek) [reply example/ck_tile/03_gemm/run_gemm_example.inc]: Good catch, thanks! Updated it also in that file

**Review by bartekxk — APPROVED** [review]

**Pull request merged**